rad/src/types/mod.rs

```diff
@@ -247,75 +321,46 @@ impl TryFrom<&[u8]> for RadonTypes {
 impl TryFrom<&RadonTypes> for Vec<u8> {
     type Error = RadError;
 
-    fn try_from(radon_types: &RadonTypes) -> Result<Vec<u8>, Self::Error> {
+    // FIXME(953): Unify all CBOR libraries
+    fn try_from(
+        radon_types: &RadonTypes,
+    ) -> Result<Vec<u8>, <Vec<u8> as TryFrom<&RadonTypes>>::Error> {
         let type_name = RadonTypes::radon_type_name(radon_types);
-        let value: Value = radon_types.clone().try_into()?;
 
-        to_vec(&value).map_err(|_| RadError::Encode {
-            from: type_name,
-            to: "Vec<u8>".to_string(),
-        })
+        match radon_types {
+            RadonTypes::RadonError(radon_error) => {
+                Vec::<u8>::try_from(radon_error).map_err(|_| RadError::Encode {
+                    from: type_name,
+                    to: "Vec<u8>".to_string(),
+                })
+            }
+            _ => {
+                let value: Value = radon_types.clone().try_into()?;
+                to_vec(&value).map_err(|_| RadError::Encode {
+                    from: type_name,
+                    to: "Vec<u8>".to_string(),
+                })
+            }
+        }
     }
 }
 
-// TODO: migrate everything to using `cbor-codec` or wait for `serde_cbor` to support CBOR tags.
+// FIXME(953): migrate everything to using `cbor-codec` or wait for `serde_cbor` to support CBOR tags.
 /// Allow decoding RADON types also from `Value` structures coming from the `cbor-codec` crate.
 /// Take into account the difference between `cbor::value::Value` and `serde_cbor::Value`.
-impl TryFrom<&cbor::value::Value> for RadonTypes {
+impl TryFrom<&CborValue> for RadonTypes {
     type Error = RadError;
 
-    fn try_from(cbor_value: &cbor::value::Value) -> Result<Self, Self::Error> {
+    fn try_from(cbor_value: &CborValue) -> Result<Self, Self::Error> {
         use cbor::value::Int;
-        use cbor::value::Value as CborValue;
 
         match cbor_value {
-            // If the tag is 39, we encode the error in a `RadError`, otherwise we ignore the tag,
-            // unbox the tagged value and decode it through recurrently calling this same function.
-            // TODO(#938): Improve CBOR tagged error handling
+            // If the tag is 39, we try to decode the value as `RadonTypes::RadonError`, otherwise
+            // we ignore the tag, unbox the tagged value and decode it through recurrently calling
+            // this same function.
             CborValue::Tagged(tag, boxed) => match (tag, std::boxed::Box::leak(boxed.clone())) {
```

Comment:
Who approved this? :rofl:

As the name implies this is a memory leak, you can probably use deref magic to get the desired value:

```suggestion
            CborValue::Tagged(tag, boxed) => match (tag, *&boxed) {
```